src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/NewAzurePolicyAssignment.cs

```diff
@@ -112,11 +137,86 @@ private JToken GetResource()
                 {
                     DisplayName = this.DisplayName ?? null,
                     PolicyDefinitionId = this.PolicyDefinition.Properties["policyDefinitionId"].Value.ToString(),
-                    Scope = this.Scope
+                    Scope = this.Scope,
+                    Parameters = this.GetParameters()
                 }
             };
 
             return policyassignmentObject.ToJToken();
         }
+
+        object IDynamicParameters.GetDynamicParameters()
+        {
+            if (this.PolicyDefinition != null)
+            {
+                var properties = this.PolicyDefinition.Properties["Properties"];
+                if (properties != null)
+                {
+                    var parameters = (properties.Value as PSObject).Properties["parameters"];
+                    if (parameters != null)
+                    {
+                        foreach (var param in (parameters.Value as PSObject).Properties)
+                        {
+                            var type = (param.Value as PSObject).Properties["type"];
+                            var typeString = type != null ? type.Value.ToString() : string.Empty;
+                            var dp = new RuntimeDefinedParameter
+                            {
+                                Name = param.Name,
+                                ParameterType = typeString.Equals("array", StringComparison.OrdinalIgnoreCase) ? typeof(object[]) : typeof(string)
+                            };
+                            dp.Attributes.Add(new ParameterAttribute
+                            {
+                                Mandatory = false,
+                                ValueFromPipelineByPropertyName = true,
+                                HelpMessage = param.Name
+                            });
+                            this.dynamicParameters.Add(param.Name, dp);
+                        }
+                    }
+                }
+            }
+
+            return this.dynamicParameters;
+        }
+
+        private JObject GetParameters()
+        {
+            // Load parameters from local file or literal
+            if (this.PolicyParameter != null)
+            {
+                string policyParameterFilePath = this.TryResolvePath(this.PolicyParameter);
+                return FileUtilities.DataStore.FileExists(policyParameterFilePath)
+                    ? JObject.Parse(FileUtilities.DataStore.ReadFileAsText(policyParameterFilePath))
+                    : JObject.Parse(this.PolicyParameter);
+            }
+
+            // Load from PS object
+            JObject parameterObject = null;
+            if (this.PolicyParameterObject != null)
+            {
+                parameterObject = new JObject();
+                foreach (string paramKey in this.PolicyParameterObject.Keys)
```

Comment:
This logic can be moved to a Util class in a method called HashTableToJToken.
Similarly you can create another Util method for the logic on line 209 DynamicParamToJToken.
If there isnt already a utility class for common Json operations like this you can create a new one and move this re-usable code there.